File: app/src/main/java/com/jcjm/desafioandroid/recycler/adapters/PullRequestAdapter.java
package com.jcjm.desafioandroid.recycler.adapters;


import android.databinding.DataBindingUtil;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.jcjm.desafioandroid.R;
import com.jcjm.desafioandroid.databinding.PullrequestItemBinding;
import com.jcjm.desafioandroid.model.PullRequest;

import java.util.ArrayList;
import java.util.List;

public class PullRequestAdapter extends RecyclerView.Adapter<PullRequestAdapter.PullRequestHolder> {


    private List<PullRequest> data;

    public PullRequestAdapter(){
        data = new ArrayList<>();
    }

    @NonNull
    @Override
    public PullRequestHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {


        PullrequestItemBinding binding = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()),
                R.layout.pullrequest_item, parent, false);

        return new PullRequestHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull PullRequestHolder holder, int position) {
        PullRequest pullRequest = data.get(position);
        holder.binding.setPull(pullRequest);

    }

    @Override
    public int getItemCount() {
        return this.data==null?0:this.data.size();
    }

    public void updateData(@Nullable List<PullRequest> data) {
        if (data == null || data.isEmpty()) {
            this.data.clear();
        } else {
            this.data.addAll(data);
        }
        notifyDataSetChanged();
    }

    public class PullRequestHolder  extends RecyclerView.ViewHolder  {

        PullrequestItemBinding binding;

        public PullRequestHolder(PullrequestItemBinding itemView) {
            super(itemView.getRoot());
            binding = itemView;
        }
    }
}

File: app/src/main/java/br/com/repository/view/adapter/PullRequestAdapter.kt
package br.com.repository.view.adapter

import android.databinding.DataBindingUtil
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import br.com.repository.R
import br.com.repository.databinding.CardviewPullRequestBinding
import br.com.repository.model.PullRequest
import br.com.repository.view.holder.PullRequestHolder

class PullRequestAdapter(private val onClick: (PullRequest) -> Unit): RecyclerView.Adapter<PullRequestHolder>() {

    private var list: List<PullRequest> = mutableListOf()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PullRequestHolder {
        val layoutInflater: LayoutInflater = LayoutInflater.from(parent.context)
        val cardviewPullRequestBinding =
                DataBindingUtil.inflate<CardviewPullRequestBinding>(layoutInflater, R.layout.cardview_pull_request,
                        parent, false)
        return PullRequestHolder(cardviewPullRequestBinding)
    }

    override fun getItemCount() = list.size

    override fun onBindViewHolder(holder: PullRequestHolder, position: Int) {
        holder.bind(list[position], onClick)
    }

    fun setListPullRequest(list: List<PullRequest>) {
        this.list = list
        notifyDataSetChanged()
    }
}